File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question2/RectangularPrism.java
/**Enfei Zhang
  March 18, 2018
  Ms. Krasteva
  This is a subclass of the abstract class Solid
 */


public class RectangularPrism extends Solid{

    private double length;
    private double height;
    private double width;

    public RectangularPrism (String n, double l, double h, double w){
        super(n);
        length = l;
        height = h;
        width = w;
    }

    @Override
    public double volume(){
        return length*height*width;
    }

    @Override
    public double surfaceArea(){
        return 2*(length*width)+2*(length*height)+2*(width*height);
    }

    @Override
    public double perimeter() {
        return 4*(length+width+height);
    }
}

File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question2/RectangularPrism.kt
/*Enfei Zhang
  March 18, 2018
  Ms. Krasteva
  This is a subclass of the abstract class Solid
 */

class RectangularPrism(n: String, private val length: Double, private val height: Double, private val width: Double) : Solid(n) {

    override fun volume(): Double {
        return length * height * width
    }

    override fun surfaceArea(): Double {
        return 2 * (length * width) + 2 * (length * height) + 2 * (width * height)
    }

    override fun perimeter(): Double {
        return 4 * (length + width + height)
    }
}
